### PR TITLE
Add filtering capabilities to Landlord Issues List

### PR DESCRIPTION
**Pull request by Rcc999**
## Changes Made
- Added a new filter panel with expandable/collapsable UI
- Implemented two filter categories:
  1. Status filter: Allows filtering by issue status (notStarted, inProgress, completed)
  2. Residence filter: Enables filtering issues by specific properties
- Added a reset filters button to quickly clear all active filters
- Maintained existing archived issues toggle while adding new filtering options
- Ensured filters can be combined (e.g., status + residence)

## Testing
Added comprehensive unit tests to verify the new filtering functionality:
- Tests for status-based filtering (notStarted, inProgress, completed)
- Tests for residence-based filtering
- Verification of filter reset functionality
- Tests for filter combinations
- UI interaction tests (expand/collapse filter panel)

## Implementation Details
- Filter state is managed using React hooks (useState)
- Filters are applied dynamically as users make selections
- Used chip-style UI components for filter options
- Maintained consistent styling with existing UI elements
- Filter selections persist until explicitly reset or changed

## Screenshots
Screenshot 2024-12-05 at 18 21 58

Screenshot 2024-12-05 at 18 22 05

## Notes
- The filtering logic preserves the existing archived toggle functionality
- Filter options are dynamically populated based on available residences

### CODE REVIEW

File: app/__tests__/unit/LandlordListIssues.test.tsx
Comment (baudoin-csp) [inline app/__tests__/unit/LandlordListIssues.test.tsx:1]: Hey @Rcc999 ! Just reviewed the maintenance issues screen test suite. It's looking pretty solid – good coverage of rendering, filtering, and interactions. We've got a great start with mocking Firebase and auth contexts. I think we could level it up by adding some more edge case tests, maybe dig into some tricky filter scenarios or error handling we might have missed. But overall, nice work on building out a robust test suite that'll help keep our code reliable.

File: package-lock.json
Comment (baudoin-csp) [inline package-lock.json:1]: Next time, please try to not push this file :)

**Review by baudoin-csp — APPROVED** [review]
PR Review Summary:

Strengths:
- Comprehensive test coverage for maintenance issues screen
- Robust mocking of Firebase and authentication contexts

Improvement Areas:
- Expand edge case tests for error scenarios and filter interactions

Detailed Notes:
- Tests validate rendering, filtering, and UI interactions
- Covers expanding/collapsing residence lists and issue filtering
- Mocking ensures reliable dependency isolation
- Opportunity to enhance test depth with more complex scenarios

Recommendation: Add more nuanced test cases to increase test suite robustness.

File: package-lock.json
Comment (Rcc999) [reply package-lock.json:1]: Okay sorry for that

**Pull request merged**